### PR TITLE
Issue Fix #11688

### LINKED ISSUE #11688: Proposal: decrease cookies usage.
Current phpmyadmin using cookies for auth type by default.
But phpmyadmin take lot's of cookies. it will use at least 10 cookies.

That's too much. because the per domain cookie limit only 20.

```
pma_lang
pma_iv-1
pma_console_mode
pma_console_height
pma_console_config
pma_collation_connection
pmaUser-1
pmaPass-1
pmaCookieVer
phpMyAdmin
```

We can decrease cookie by merge cookie into json style value or other ways to do this.

2 - 3 cookies will be good.

### PR DESCRIPTION
**Pull request by flash1452**
Decrease Cookie Usage by modifying existing structure.
Signed-Off-by:Vishal Pandey vishpandey2014@gmail.com

### CODE REVIEW

**Comment by flash1452** [comment]
@nijel The Travis CI build is failing everytime i make a PR.The report says "command ant lint exited with 1".Can you provide a solution for this error.

**Comment by flash1452** [comment]
@nijel It would be of great help if you can test my PR and guide me where am going wrong.

**Comment by Achilles-96** [comment]
@flash1452 Please use --signoff while you are committing.

File: libraries/plugins/auth/AuthenticationCookie.php [hunk 1]
```diff
@@ -317,17 +317,17 @@ public function authCheck()
             // -> delete password cookie(s)
             if ($GLOBALS['cfg']['LoginCookieDeleteAll']) {
                 foreach ($GLOBALS['cfg']['Servers'] as $key => $val) {
-                    $GLOBALS['PMA_Config']->removeCookie('pmaPass-' . $key);
-                    if (isset($_COOKIE['pmaPass-' . $key])) {
-                        unset($_COOKIE['pmaPass-' . $key]);
+                    $GLOBALS['PMA_Config']->removeCookie('pma_auth');
+                    if (isset($_COOKIE['pma_auth'])) {
+                        unset($_COOKIE['pma_auth']);
```
Comment (nijel) [inline libraries/plugins/auth/AuthenticationCookie.php]: Why did you remove the server index here?

**Comment by nijel** [comment]
You've broken the tests, that's why the checks are failing. 
Also you would be great if you could squash commits.

File: libraries/plugins/auth/AuthenticationCookie.php (hunk 1 above)
Comment (flash1452) [inline libraries/plugins/auth/AuthenticationCookie.php]: @nijel The reason behind removing the server index is that if i have a situation where i have to remove password cookie,according to the new structure i will have to remove the whole json cookie array.If there is any workaround that you can suggest me,it would be of great help.

File: js/console.js [hunk 2]
```diff
@@ -64,12 +64,18 @@ var PMA_console = {
...
+
         }
-        if (! $.cookie('pma_console_mode')) {
-            $.cookie('pma_console_mode', 'info');
+        if (! $.cookie('console_info')) {
+            console_info.pma_console_mode="info";
+
         }
+        $.cookie('console_info', JSON.stringify(console_info));
+        //console.log(JSON.parse($.cookie('pma_data')));
```
Comment (ibennetch) [inline js/console.js]: Is this debugging code that can be removed?

**Comment by ibennetch** [comment]
The Codacy report is pretty self-explanatory and sometimes not related to your changes, so for now I'll focus on the Travis tests which give several errors.

A lot of the errors appear to be with testing the cookie data, for instance

> [exec] 1) AuthenticationCookieTest::testAuthCheckDecryptUser
> [exec] Undefined index: pma_data

and

> [exec] 3) AuthenticationCookieTest::testAuthCheckAuthFails
> [exec] Undefined index: pma_data

So in the test file the test has to be adjusted for the new style cookies. Those were the ones that jumped out to me,

File: js/console.js (hunk 2 above)
Comment (flash1452) [inline js/console.js]: NO sir,its not a debugging code it actually sets the initial values of console_mode as "info" and console_height as 92 in the cookie,which is further accessed to change the value of console_mode to "collapse"

Comment (nijel) [inline js/console.js]: And  what the commented out console.log line?

**Comment by nijel** [comment]
Few general notes to Git usage:

- use menaningful commit mesages, imagine somebody looking at your changes in a year and he seems "test_file_changes" commit message
- include Signed-Off-By directly in commit message (use --signoff parameter of git if using command line)
- squash commits as most of them really do not have to be separate and are just fixups to previous code
